toothpick-runtime/src/main/java/toothpick/UnScopedProviderImpl.java

```diff
@@ -44,60 +46,65 @@ public UnScopedProviderImpl(Class<?> factoryKeyClass, boolean isProviderFactoryC
 
   //we lock on the unbound provider itself to prevent concurrent usage
   //of the unbound provider (
-  public synchronized T get(Scope scope) {
-    if (instance != null) {
-      return instance;
-    }
-
-    if (providerInstance != null) {
-      if (isLazy) {
-        instance = providerInstance.get();
-        //gc
-        providerInstance = null;
+  public T get(Scope scope) {
+    lockGet.lock();
```

Comment:
They are just faster. But I agree, it's a very minor change here.
http://lycog.com/concurency/performance-reentrantlock-synchronized/